scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -1928,10 +1928,20 @@ class EDT:
     compat2nodes:
       A collections.defaultdict that maps each 'compatible' string that appears
       on some Node to a list of Nodes with that compatible.
+      The raw indexes of the nodes in these lists are used for node instance
+      numbering within a compatible. The nodes within a compatible are sorted
+      by enabled-status. This is a way to satisfy peoples intuitions about instance
+      numbers. It also ensures that drivers and applications which do not use
+      instance numbers can use their singleton with index 0.
+      This doesn't violate any devicetree.h API guarantees about instance ordering,
+      since we make no promises that instance numbers are stable across builds.
```

Comment:
as mentioned in other comment, the devicetree spec doesn't have the concept of instance numbers, nor references to any header files, therefore it appears wrong to me to refer to instance numbers or header files in edtlib.py.